Add configuration for empirically-corrected ozone parameters

DWesl opened the pull request:
<!--
  *** PLEASE READ ***

  Any PRs not following this template will be closed.

  Please delete all these comments before submitting the PR.

  Please use a short (<60 char), descriptive title for the PR title above. It should complete the sentence "If merged, this PR will _____". Capitalize the first word and do not end with a period.

  No content should appear above the "Description" header.

  If this PR is not merge-ready (e.g. it depends on other PRs not yet merged), please mark it as draft until it is ready.

  PRs should meet these guidelines:
  - Each PR should address ONE topic and have an associated issue.
  - No hard-coded paths or personal directories.
  - No temporary or backup files should be committed (including logs).
  - Any code that you disabled by being commented out should be removed or reenabled.
-->
# Description
<!-- This description will become the commit message for the PR. -->
<!--
  Solely pointing to an issue is not an adequate description!

  Please use this format for your description:

  Describe your changes. Focus on the *what* and *why*. The *how* will be evident from the changes. In particular, be sure to note any interface changes, such as command line syntax, that will need to be communicated to users.

  At the end of your description, please be sure to add the issue this PR solves using the word "Resolves". If there are any issues that are related but not yet resolved (including in other repos), you may use "Refs".

  Resolves #1234
  Refs #4321
  Refs NOAA-EMC/repo#5678
-->
The current CCPP ozone parametrization, based on [McCormack et al. (2006)](https://acp.copernicus.org/articles/6/4943/2006/), does not handle the Antarctic ozone hole, as noted in that paper.  This PR uses ozone loss profiles from [Hassler et al. (2011, JGR-A)](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2011JD016353#jgrd17439-fig-0005) to update the climatological or reference ozone net production term in the parameter files, with tweaks for better agreement with GDAS column ozone values.

This PR requires a new file in `FIXgfs/am` or `HOMEgfs/fix/am`, that file being available in [modify-ozone-production-loss.zip](https://github.com/user-attachments/files/18970267/modify-ozone-production-loss.zip) as "new-params/ozprdlos_2015_new_sbuvO3_tclm15_nuchem_shozhvlogp.f77", together with the program to modify the existing file, in "new-params/daylight_hours.f90" and "new-params/modify_ozprdloss_file.f90"
These files are available on Hera at `/scratch1/NCEPDEV/global/Daniel.Wesloh/global-workflow-20250128/new-params/ozprdlos_2015_new_sbuvO3_tclm15_nuchem_shozhvlogp.f77`

Resolves #2768 
<!-- For more on writing good commit messages, see https://cbea.ms/git-commit/ -->

# Type of change
- [ ] Bug fix (fixes something broken)
- [X] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [ ] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 -->
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [ ] UFS-weather-model <!-- ufs-community/ufs-weather-model#1234 -->
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?
<!-- Please list any test you conducted, including the machine.

Example:
- Clone and build on WCOSS
- Cycled test on Orion
- Forecast-only on Hera
-->
Forecast-only tests on Hera, documented in [a presentation to the Physics group](https://docs.google.com/presentation/d/1hySHCZuvu8YvX4D9KCXtNYIxLMSPxX_MzRVZoL7yfFA/edit?usp=sharing)
Five years of ATM runs at C192, twenty years coupled at C96, and sixteen sixteed-day cases at C768

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

DWesl committed:
ENH: Add configuration for empirically-corrected ozone parameters.

XiaqiongZhou-NOAA reviewed (commented):
It seems not good to add this new parameter in config.base. It is better to add it to config.fcst and config.ecf. These two file should be copied from parm/config/gefs instead of soft link and then add the new parameter.

DWesl committed:
Move ozone parameter chooser from config.base to config.fcst

DWesl committed:
Move ozone parameter chooser from config.base to config.fcst and config.efcs

DWesl commented:
`config.{fcst,efcs}` copied from `gefs`; new configuration variable added to end of each file and removed from `config.base`

aerorahul reviewed (dismissed):
looks good.

aerorahul committed:
Merge branch 'develop' into empirical-ozone-correction

aerorahul commented:
@KateFriedman-NOAA 
This PR requires new files to be staged.  When you are done staging on all machines, please merge this PR.

KateFriedman-NOAA commented:
@DWesl I copied the new `fix/am` `ozprdlos_2015_new_sbuvO3_tclm15_nuchem_shozhvlogp.f77` file into the primary fix set on Hera. Please confirm this meets the request and then I'll sync it to the other platforms and merge this PR. Thanks!
```
20:10:44 20220805/$pwd
/scratch1/NCEPDEV/global/glopara/fix/am/20220805
20:10:48 20220805/$ll ozprdlos_2015_new_sbuvO3_tclm15_nuchem*
-rw-r--r-- 1 role.glopara global 1114844 Mar  8  2020 ozprdlos_2015_new_sbuvO3_tclm15_nuchem.f77
-rw-r--r-- 1 role.glopara global 1300564 Jan 28 19:35 ozprdlos_2015_new_sbuvO3_tclm15_nuchem_shozhvlogp.f77
```

DWesl commented on ush/forecast_predet.sh:
```diff
@@ -499,7 +499,11 @@ FV3_predet(){
   if [[ "${new_o3forc:-YES}" == "YES" ]]; then
-    O3FORC="ozprdlos_2015_new_sbuvO3_tclm15_nuchem.f77"
+    if [[ "${o3forc_params:-McCormack}" == "McCormack-empirical-sh-ozh" ]]; then
+      O3FORC="ozprdlos_2015_new_sbuvO3_tclm15_nuchem_zhozhvlogp.f77"
```
```suggestion
      O3FORC="ozprdlos_2015_new_sbuvO3_tclm15_nuchem_shozhvlogp.f77"
```

DWesl committed:
FIX: Use correct name for new parameter file

It's intended to be mnemonic:
Southern Hemisphere OZone Hole, latitudinal extent determined by equivalent latitude of the polar Vortex and vertical profile calculated in terms of LOG-Pressure.

DWesl commented:
The file checksums match and I fixed the file name in the PR so that also matches: looks good to me

aerorahul reviewed (approved):
looks good.
I assume the removal of the soft-links and the committing of `config.fcst` and `config.efcs` are due to the new option:  `export o3forc_params="McCormack-empirical-sh-ozh"`.
The rest are the same.

KateFriedman-NOAA commented:
> The file checksums match and I fixed the file name in the PR so that also matches: looks good to me

Thanks for checking and confirming @DWesl ! I have synced the new fix file to WCOSS2, MSU, Gaea C6, AWS, and HPSS.

DWesl commented:
> I assume the removal of the soft-links and the committing of `config.fcst` and `config.efcs` are due to the new option: `export o3forc_params="McCormack-empirical-sh-ozh"`

Correct.  Would you prefer I sourced the old files rather than copying them?

KateFriedman-NOAA reviewed (approved):
Looks good, thanks @DWesl !

aerorahul commented:
> Correct. Would you prefer I sourced the old files rather than copying them?

No.  This is fine.  When they were first added to `parm/config/sfs`, SFS was assumed to have the GEFS model configuration.  That is no longer true and maintaining SFS-specific configuration is fine.

aerorahul committed:
Merge branch 'develop' into empirical-ozone-correction
